src/python/pants/backend/python/lint/autoflake/rules.py

```diff
@@ -48,21 +59,27 @@ async def autoflake_fmt(request: AutoflakeRequest, autoflake: Autoflake) -> FmtR
             argv=(
                 "--in-place",
                 *autoflake.args,
-                *request.snapshot.files,
+                *snapshot.files,
             ),
-            input_digest=request.snapshot.digest,
-            output_files=request.snapshot.files,
-            description=f"Run Autoflake on {pluralize(len(request.field_sets), 'file')}.",
+            input_digest=snapshot.digest,
+            output_files=snapshot.files,
+            description=f"Run Autoflake on {pluralize(len(request.files), 'file')}.",
             level=LogLevel.DEBUG,
         ),
     )
     output_snapshot = await Get(Snapshot, Digest, result.output_digest)
-    return FmtResult.create(request, result, output_snapshot, strip_chroot_path=True)
+    return FmtResult.create(
+        result,
+        snapshot,
+        output_snapshot,
+        strip_chroot_path=True,
+        formatter_name=AutoflakeRequest.name,
+    )
 
 
 def rules():
     return [
         *collect_rules(),
-        UnionRule(FmtTargetsRequest, AutoflakeRequest),
+        *AutoflakeRequest.registration_rules(),
```

Comment:
Could probably just be `.rules()`? Aligns with https://github.com/pantsbuild/pants/pull/16954

Reply:
Hmm that'll have to be in a follow up PR. But +1.